### PR TITLE
Turn `@pybind11` into an internal dependency.

### LINKED ISSUE #71: move to rules_python and remove local_config_python
Although https://github.com/bazelbuild/rules_python/issues/824 isn't formally "done" yet, it seems that pybind11_bazel can move wholesale to rules_python (i.e. using `@rules_python//python/cc:current_py_cc_headers` and `@rules_python//python/cc:current_py_cc_libs`) and remove local_config_python entirely. At that point, the name "python_configure" will no longer make sense and should be replaced by, say, "pybind11_configure". Notably, only Bzlmod users need to "configure" something (just getting pybind11 itself) whereas `WORKSPACE` users no longer need to "configure" anything.

### PR DESCRIPTION
**Pull request by junyer**
Fixes #71.

### CODE REVIEW

**Comment by junyer** [comment]
PTAL, @rickeylev. (FYI, @rwgk and @david-crouse.)

File: BUILD [hunk 1]
```diff
@@ -1,4 +1,20 @@
 # Bazel extensions for pybind11
+package(default_visibility = ["//visibility:public"])
+
 licenses(["notice"])
 
 exports_files(["LICENSE"])
+
+# Export `@pybind11` targets in order to make them available to users.
+# In particular, `build_defs.bzl` uses `pybind11` and `msvc_compiler`.
```
Comment (rickeylev) [inline BUILD:9]: > In particular, `build_defs.bzl` uses `pybind11` and `msvc_compiler`.

I'm guessing you ran into the problem where a user of pybind11_bazel calls the pybind11_bazel macro, which uses the string `@pybind11//:bla`, but that repo isn't visible to the user's build file? You can work around this by calling `Label()` in the context (bzl file) where the repo is visible.

i.e.

```
PYBIND_COPTS = select({
  Label("@pybind11//:msvc_compiler"): [],
})

PYBIND_DEPS = [
  Label(""@pybind11//:pybind11"),
]
```

The Label() call makes bazel convert the apparent name ("@pybind11") to the canonical name ("@@pybind11~whatever~balbal") (apparent and canonical are the bazel terms used to describe these two different "types" of strings)

I think that makes these aliases unnecessary? Or do some of them still need to be public things users might want to explicitly use themselves for other reasons?

File: BUILD [hunk 2]
```diff
@@ -1,4 +1,20 @@
 # Bazel extensions for pybind11
+package(default_visibility = ["//visibility:public"])
```
[the remark this answers is not in the document]
Comment (junyer) [reply BUILD:2]: Done.

File: BUILD [hunk 3]
```diff
@@ -1,4 +1,20 @@
...
 exports_files(["LICENSE"])
+
+# Export `@pybind11` targets in order to make them available to users.
+# In particular, `build_defs.bzl` uses `pybind11` and `msvc_compiler`.
+[
+    alias(
+        name = name,
+        actual = "@pybind11//:{}".format(name),
+    )
+    for name in [
```
[the remark this answers is not in the document]
Comment (junyer) [reply BUILD:15]: Obsoleted as per the discussion below. :)

File: BUILD (hunk 1 above)
Comment (junyer) [reply BUILD:9]: TIL! Thank you! :D

I'm going to go ahead and yeet the `pybind11_embed` target **_now_** then, which means that `BUILD` just remains empty.

**Review by rickeylev — APPROVED** [review]

**Pull request merged**